stacks-common/src/util/vrf.rs

```diff
@@ -149,71 +114,76 @@ impl Default for VRFPrivateKey {
 impl VRFPrivateKey {
     pub fn new() -> VRFPrivateKey {
         let mut rng = rand::thread_rng();
-        let keypair: VRFKeypair = VRFKeypair::generate(&mut rng);
-        VRFPrivateKey(keypair.secret)
+        let signing_key = ed25519_dalek::SigningKey::generate(&mut rng);
+        VRFPrivateKey(signing_key)
     }
 
     pub fn from_hex(h: &str) -> Option<VRFPrivateKey> {
-        match hex_bytes(h) {
-            Ok(b) => match ed25519_PrivateKey::from_bytes(&b[..]) {
-                Ok(pk) => Some(VRFPrivateKey(pk)),
-                Err(_) => None,
-            },
-            Err(_) => None,
-        }
+        let bytes = hex_bytes(h).ok()?;
+        Self::from_bytes(bytes.as_slice())
     }
 
     pub fn from_bytes(b: &[u8]) -> Option<VRFPrivateKey> {
-        match ed25519_PrivateKey::from_bytes(b) {
-            Ok(pk) => Some(VRFPrivateKey(pk)),
-            Err(_) => None,
-        }
+        let signing_key = ed25519_dalek::SigningKey::try_from(b).ok()?;
+        Some(VRFPrivateKey(signing_key))
     }
 
     pub fn to_hex(&self) -> String {
         to_hex(self.as_bytes())
     }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        self.0.as_bytes()
+    }
+
+    pub fn to_bytes(&self) -> [u8; 32] {
+        self.0.to_bytes()
+    }
 }
 
 impl VRFPublicKey {
-    pub fn from_private(pk: &VRFPrivateKey) -> VRFPublicKey {
-        VRFPublicKey(ed25519_PublicKey::from(&pk.0))
+    pub fn from_private(sk: &VRFPrivateKey) -> VRFPublicKey {
+        VRFPublicKey(sk.0.verifying_key())
     }
 
+    /// Verify that a given byte string is a well-formed EdDSA public
+    /// key (i.e. it's a compressed Edwards point that is valid), and return
+    /// a VRFPublicKey if so
     pub fn from_bytes(pubkey_bytes: &[u8]) -> Option<VRFPublicKey> {
```

Comment:
I know you didn't introduce this function in this PR, but this really should be `try_from_bytes()`, or even better, `impl TryFrom<&[u8]> for VRFPublicKey`, and return a `Result`

Reply:
Yeah, I agree, but I don't want to do too much refactoring in this PR (and this would mean a pretty fair amount of refactoring).